android/src/main/java/com/op/sqlite/OPSQLiteModule.kt

```diff
@@ -47,25 +50,33 @@ internal class OPSQLiteModule(context: ReactApplicationContext?) : ReactContextB
         }
     }
 
-    @ReactMethod(isBlockingSynchronousMethod = true)
-    fun moveAssetsDatabase(name: String, extension: String): Boolean {
+    @ReactMethod
+    fun moveAssetsDatabase(args: ReadableMap, promise: Promise) {
+        val filename = args.getString("filename")!!
+        val path = args.getString("path") ?: "custom"
+        val overwrite = if(args.hasKey("overwrite")) { args.getBoolean("overwrite") } else false
         val context = reactApplicationContext
         val assetsManager = context.assets
 
         try {
             // Open the input stream for the asset file
-            val inputStream: InputStream = assetsManager.open("custom/$name.$extension")
+            val inputStream: InputStream = assetsManager.open("$path/$filename")
 
             // Create the output file in the documents directory
             val databasesFolder =
                     context.getDatabasePath("defaultDatabase")
                             .absolutePath
                             .replace("defaultDatabase", "")
 
-            val outputFile = File(databasesFolder, "$name.$extension")
+            val outputFile = File(databasesFolder, filename)
 
             if (outputFile.exists()) {
```

Comment:
Would it not be more efficient to check if the output file exists before even opening the input stream, so that it is only opened if we will actually use it? This doesn't matter if we are overwriting the file, but in the case where we're not and the file already exists, it would make a difference.

Reply:
Don't think opening a stream makes such a difference, but I will move it.